File: src/main/java/tn/esprit/nafseyti/models/Question.java
package tn.esprit.nafseyti.models;

import java.sql.Timestamp;

public class Question {
    private int id;
    private int testId;
    private String testNom; // For display purposes
    private String texte;
    private String typeQuestion;
    private String reponsesPossibles;
    private int points;
    private int ordre;
    private boolean obligatoire;
    private String status;
    private Timestamp createdAt;

    // Constructors
    public Question() {
        this.points = 1;
        this.ordre = 1;
        this.obligatoire = true;
        this.status = "actif";
    }

    public Question(int id, int testId, String texte, String typeQuestion,
                    String reponsesPossibles, int points, int ordre,
                    boolean obligatoire, String status, Timestamp createdAt) {
        this.id = id;
        this.testId = testId;
        this.texte = texte;
        this.typeQuestion = typeQuestion;
        this.reponsesPossibles = reponsesPossibles;
        this.points = points;
        this.ordre = ordre;
        this.obligatoire = obligatoire;
        this.status = status;
        this.createdAt = createdAt;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getTestId() {
        return testId;
    }

    public void setTestId(int testId) {
        this.testId = testId;
    }

    public String getTestNom() {
        return testNom;
    }

    public void setTestNom(String testNom) {
        this.testNom = testNom;
    }

    public String getTexte() {
        return texte;
    }

    public void setTexte(String texte) {
        this.texte = texte;
    }

    public String getTypeQuestion() {
        return typeQuestion;
    }

    public void setTypeQuestion(String typeQuestion) {
        this.typeQuestion = typeQuestion;
    }

    public String getReponsesPossibles() {
        return reponsesPossibles;
    }

    public void setReponsesPossibles(String reponsesPossibles) {
        this.reponsesPossibles = reponsesPossibles;
    }

    public int getPoints() {
        return points;
    }

    public void setPoints(int points) {
        this.points = points;
    }

    public int getOrdre() {
        return ordre;
    }

    public void setOrdre(int ordre) {
        this.ordre = ordre;
    }

    public boolean isObligatoire() {
        return obligatoire;
    }

    public void setObligatoire(boolean obligatoire) {
        this.obligatoire = obligatoire;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    @Override
    public String toString() {
        return "Question{" +
                "id=" + id +
                ", testId=" + testId +
                ", texte='" + texte + '\'' +
                ", typeQuestion='" + typeQuestion + '\'' +
                ", points=" + points +
                ", ordre=" + ordre +
                ", obligatoire=" + obligatoire +
                ", status='" + status + '\'' +
                '}';
    }
}

File: src/Entity/Question.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\QuestionRepository;

#[ORM\Entity(repositoryClass: QuestionRepository::class)]
#[ORM\Table(name: 'questions')]
class Question
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): self
    {
        $this->id = $id;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $test_id = null;

    public function getTest_id(): ?int
    {
        return $this->test_id;
    }

    public function setTest_id(int $test_id): self
    {
        $this->test_id = $test_id;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: false)]
    private ?string $texte = null;

    public function getTexte(): ?string
    {
        return $this->texte;
    }

    public function setTexte(string $texte): self
    {
        $this->texte = $texte;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $type_question = null;

    public function getType_question(): ?string
    {
        return $this->type_question;
    }

    public function setType_question(?string $type_question): self
    {
        $this->type_question = $type_question;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $reponses_possibles = null;

    public function getReponses_possibles(): ?string
    {
        return $this->reponses_possibles;
    }

    public function setReponses_possibles(?string $reponses_possibles): self
    {
        $this->reponses_possibles = $reponses_possibles;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $points = null;

    public function getPoints(): ?int
    {
        return $this->points;
    }

    public function setPoints(?int $points): self
    {
        $this->points = $points;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $ordre = null;

    public function getOrdre(): ?int
    {
        return $this->ordre;
    }

    public function setOrdre(?int $ordre): self
    {
        $this->ordre = $ordre;
        return $this;
    }

    #[ORM\Column(type: 'boolean', nullable: true)]
    private ?bool $obligatoire = null;

    public function isObligatoire(): ?bool
    {
        return $this->obligatoire;
    }

    public function setObligatoire(?bool $obligatoire): self
    {
        $this->obligatoire = $obligatoire;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $status = null;

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(?string $status): self
    {
        $this->status = $status;
        return $this;
    }

    #[ORM\Column(type: 'datetime', nullable: false)]
    private ?\DateTimeInterface $created_at = null;

    public function getCreated_at(): ?\DateTimeInterface
    {
        return $this->created_at;
    }

    public function setCreated_at(\DateTimeInterface $created_at): self
    {
        $this->created_at = $created_at;
        return $this;
    }

}
